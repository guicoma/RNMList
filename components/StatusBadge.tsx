import { CharacterStatus } from "@/types";
import { StyleSheet, Text, useColorScheme, View } from "react-native";
import { Colors } from "@/constants/Colors";

export default function StatusBadge({status}: {status: CharacterStatus}) {
    const theme = useColorScheme() ?? 'light';
    const statusColor = Colors[theme].badge;

    return (
        <View style={[styles.badge,{backgroundColor: statusColor[status]}]}>
            <Text style={styles.text}>{status}</Text>
        </View>
    )
}

const styles = StyleSheet.create({
    badge: {
        alignItems  : 'baseline',
        flexDirection: 'row',
        borderRadius: 6,
        paddingVertical: 2,
        paddingHorizontal: 6,
    },
    text: {
        color: 'white'
    }
})