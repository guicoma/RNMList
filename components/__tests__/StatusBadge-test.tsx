import * as React from 'react';
import renderer from 'react-test-renderer';

import StatusBadge from '../StatusBadge';

it(`renders correctly`, () => {
  const tree = renderer.create(<StatusBadge status={'Alive'} />).toJSON();

  expect(tree).toMatchSnapshot();
});
