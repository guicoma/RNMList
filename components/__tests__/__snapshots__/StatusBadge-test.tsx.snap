// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "baseline",
        "borderRadius": 6,
        "flexDirection": "row",
        "paddingHorizontal": 6,
        "paddingVertical": 2,
      },
      {
        "backgroundColor": "#08C952",
      },
    ]
  }
>
  <Text
    style={
      {
        "color": "white",
      }
    }
  >
    Alive
  </Text>
</View>
`;
